api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirAllergyIntoleranceServiceImplTest.java

```diff
@@ -276,4 +279,47 @@ public void saveAllergy_shouldSaveNewAllergy() {
 		assertThat(result.getId(), equalTo(ALLERGY_UUID));
 	}
 	
+	@Test(expected = InvalidRequestException.class)
+	public void updateAllergy_shouldThrowInvalidRequestExceptionIfIdIsNull() {
+		AllergyIntolerance allergyIntolerance = new AllergyIntolerance();
+		allergyIntolerance.setId(ALLERGY_UUID);
+		
+		AllergyIntolerance result = service.update(null, allergyIntolerance);
+	}
+	
+	@Test(expected = InvalidRequestException.class)
+	public void updateAllergy_shouldThrowInvalidRequestException() {
```

Comment:
I'd prefer `updateAllergy_shouldThrowInvalidRequestExceptionWhenResourceIdDoesntMatchUrl` or something so it's clear what's being tested (yes its verbose)